### PR TITLE
[MVC-1107] Limit number of visible reports and show most recent first

### PR DESCRIPTION
**Pull request by nashthecoder**
[MVC-1107] Limit number of visible reports and show most recent first
Success:
1. Created annual and financial report collections - moved them from uploads 
2. Able to limit files to 2 per category

Challenges: 
1. Need make the  pdf link clickable
2. Need to sort files in descending order. 
2. The report cover files are an issue when it comes to formatting to fit report layout. 

[MVC-1181]: https://triggerise.atlassian.net/browse/MVC-1181

[MVC-1107]: https://triggerise.atlassian.net/browse/MVC-1107

### CODE REVIEW

**Comment by nashthecoder** [comment]
@tiagovarela I may need your help on the challenges.

**Comment by raguiar9080** [comment]
@nashthecoder Haven't really looked at current solution but regarding the challenges:
>- Need make the pdf link clickable

Isn't a link always clickable? Not sure I understand here

>- Need to sort files in descending order.

I think this should probably be done via JS at startup no?

>- The report cover files are an issue when it comes to formatting to fit report layout.

I'ma. bit lost on this one sorry 😅 Is this inside a report? Or the layout on the page?

**Comment by nashthecoder** [comment]
Screenshot 2021-05-28 at 11 02 03
 Fixed alignment and grid issues.

**Comment by nashthecoder** [comment]
> @nashthecoder Haven't really looked at current solution but regarding the challenges:
> 
> > * Need make the pdf link clickable 
> 
> Isn't a link always clickable? Not sure I understand here - not by default. 
> 
> > * Need to sort files in descending order.
> 
> I think this should probably be done via JS at startup no? - Thanks for the tip. 
> 
> > * The report cover files are an issue when it comes to formatting to fit report layout.
> 
> I'ma. bit lost on this one sorry 😅 Is this inside a report? Or the layout on the page?   - these are the icons on the cover page.

**Comment by nashthecoder** [comment]
Video of Report view on page load and with filters applied. 
https://user-images.githubusercontent.com/15178823/120630720-c3c7d780-c46f-11eb-908d-7cd80d7df0ed.mov

On page load image: 
Screenshot 2021-06-03 at 13 31 15
Filters Applied 
Screenshot 2021-06-03 at 13 31 48
Screenshot 2021-06-03 at 13 31 40
Screenshot 2021-06-03 at 13 31 29

File: assets/main.js [hunk 1]
```diff
@@ -252,26 +252,30 @@ function classLists() {
...
 function filters(){
- for(i=0; i<filtersCheck.length; i++) {
-   filtersCheck[i].addEventListener('click', function(){
-     document.querySelectorAll('.report-cover').forEach((item, i) => {
-       item.classList.remove('display');
-     });
-
+  var allReports = Array.from(document.querySelectorAll('.report-cover'))
+  hideReports(allReports)
+  console.log(allReports);
```
Comment (raguiar9080) [inline assets/main.js]: I don't think this is intended on prod ✂️

**Review by raguiar9080 — APPROVED** [review]
Solution is really clean, nice job 👍 
There seems to still be a conflict on `_config.yml` but other than that :shipit:

File: assets/main.js (hunk 1 above)
Comment (nashthecoder) [reply assets/main.js]: @raguiar9080 you mean the console.log?

Comment (raguiar9080) [reply assets/main.js]: @nashthecoder yes

**Pull request merged**